fontforge/woff.c

```diff
@@ -348,13 +348,45 @@ return( NULL );
     }
 
     if ( sf!=NULL && metaOffset!=0 ) {
-	char *temp = malloc(metaLenCompressed+1);
+	/*
+	* Boundary/integer overflow checks:
+	*
+	* We don't want to actually dereference a null pointer (returned
+	* by asking to allocate too much RAM) and we don't want to allocate
+	* a 0-sized chunk (caused when one of the (metaLenxxx + 1) values overflows).
+	*
+	* We can safely pass sf->woffMetadata as a NULL pointer because
+	* it's never accessed anywhere else without a check for it being
+	* NULL first
+	*/
+	if(metaLenUncompressed == (uint32_t)0xffffffff) {
```

Comment:
Right. Well I guess if we use UINT32_MAX we can do away with the cast because presumably it's already well defined for the system it's being compiled on.

P.S Does anyone know why FF redefines the standard types such as `uint32` (in basics.h) instead of just using `uint32_t`?